### ISSUE TITLE
V3 - Industrial Communications Activities

### ISSUE THREAD

**Issue by KnibbsyMan** [elsa-workflows/elsa-core#4620]
I work in the manufacturing industry and have been looking at using Elsa Workflows to help simplify some of the repetitive tasks we have to do and to automate certain activities. My question is, would including support for industrial ethernet communications protocols, i.e. OPC UA or Modbus, be within the scope of what you are doing with the project? I can see there is already a growing selection of activities types, so I was wondered if this is something you would consider adding to the project? Being able to ingest data directly from shop floor events or tag value changes would be really useful in some of my use cases. 

An example of this could be, when the temperature of a  sensors value goes above X, log it in a database and send an email to management to warn them of the issue.

**Comment by sfmskywalker** [comment]
Absolutely, your suggestion for integrating such protocols into Elsa is both insightful and relevant, especially for applications in the manufacturing sector. I envision Elsa as a platform that supports a wide range of activities, including those specific to industrial automation.

While these specialized protocols may be outside the immediate expertise of the core Elsa team, we highly encourage and support community contributions to extend our functionalities. Modules like the ones you propose could ideally be developed and maintained in a community repository. This collaborative approach allows experts like yourself to contribute specific knowledge, enriching the Elsa ecosystem.

In the future, we might establish a marketplace website for easier discovery of these modules. This could include both free and premium modules, with premium modules potentially residing in vendor-specific repositories, like Elsa Digital (the commercial branch around Elsa), which will contain curated, high-quality modules. Community-contributed modules will be housed in a separate "community" repository.

We're keen on fostering a collaborative environment where users from various industries can contribute. Your use case of triggering actions based on sensor data is a great example of the potential applications of Elsa in industrial settings.
